MIDDLEWARE_CLASSES setting is deprecated in Django 1.10

```
MIDDLEWARE_CLASSES += [
    # 'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.locale.LocaleMiddleware',
    # 'django.middleware.common.CommonMiddleware',
    # 'django.middleware.csrf.CsrfViewMiddleware',
    # 'django.contrib.auth.middleware.AuthenticationMiddleware',
    # 'django.contrib.messages.middleware.MessageMiddleware',
    # 'django.middleware.clickjacking.XFrameOptionsMiddleware',

    'spirit.core.middleware.XForwardedForMiddleware',
    'spirit.user.middleware.TimezoneMiddleware',
    'spirit.user.middleware.LastIPMiddleware',
    'spirit.user.middleware.LastSeenMiddleware',
    'spirit.user.middleware.ActiveUserMiddleware',
    'spirit.core.middleware.PrivateForumMiddleware',
]
```
If I change `MIDDLEWARE_CLASSES` above to `MIDDLEWARE` errors occur:
```
[2017-01-09 20:35:10 +0800] [9759] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/home/i/Desktop/env/lib/python3.5/site-packages/gunicorn/arbiter.py", line 557, in spawn_worker
    worker.init_process()
  File "/home/i/Desktop/env/lib/python3.5/site-packages/gunicorn/workers/base.py", line 126, in init_process
    self.load_wsgi()
  File "/home/i/Desktop/env/lib/python3.5/site-packages/gunicorn/workers/base.py", line 136, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/home/i/Desktop/env/lib/python3.5/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/home/i/Desktop/env/lib/python3.5/site-packages/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/home/i/Desktop/env/lib/python3.5/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/home/i/Desktop/env/lib/python3.5/site-packages/gunicorn/util.py", line 357, in import_app
    __import__(module)
  File "/home/i/Desktop/mysite/myapp/wsgi.py", line 16, in <module>
    application = get_wsgi_application()
  File "/home/i/Desktop/env/lib/python3.5/site-packages/django/core/wsgi.py", line 14, in get_wsgi_application
    return WSGIHandler()
  File "/home/i/Desktop/env/lib/python3.5/site-packages/django/core/handlers/wsgi.py", line 153, in __init__
    self.load_middleware()
  File "/home/i/Desktop/env/lib/python3.5/site-packages/django/core/handlers/base.py", line 82, in load_middleware
    mw_instance = middleware(handler)
TypeError: object() takes no parameters
[2017-01-09 20:35:10 +0800] [9759] [INFO] Worker exiting (pid: 9759)
[2017-01-09 20:35:10 +0800] [9754] [INFO] Shutting down: Master
[2017-01-09 20:35:10 +0800] [9754] [INFO] Reason: Worker failed to boot.
```
So now I can't keep both `MIDDLEWARE_CLASSES` and `MIDDLEWARE`, and have to use `MIDDLEWARE_CLASSES` only.
Now I can see the forum and login page.

--- a/spirit/core/middleware.py
+++ b/spirit/core/middleware.py
@@ -6,15 +6,20 @@ from django.conf import settings
 from django.contrib.auth.views import redirect_to_login
 from django.core.urlresolvers import resolve
 
+try:
+    from django.utils.deprecation import MiddlewareMixin
+except ImportError:  # Django < 1.10
+    MiddlewareMixin = object
 
-class XForwardedForMiddleware(object):
+
+class XForwardedForMiddleware(MiddlewareMixin):
 
     def process_request(self, request):
         if not settings.DEBUG:
             request.META['REMOTE_ADDR'] = request.META['HTTP_X_FORWARDED_FOR'].split(",")[-1].strip()
 
 
-class PrivateForumMiddleware(object):
+class PrivateForumMiddleware(MiddlewareMixin):
 
     def process_request(self, request):
         if not settings.ST_PRIVATE_FORUM:

--- a/spirit/user/middleware.py
+++ b/spirit/user/middleware.py
@@ -9,6 +9,11 @@ from django.conf import settings
 from django.contrib.auth import logout
 from django.utils import timezone
 
+try:
+    from django.utils.deprecation import MiddlewareMixin
+except ImportError:  # Django < 1.10
+    MiddlewareMixin = object
+
 from .models import UserProfile
 
 
@@ -21,7 +26,7 @@ __all__ = [
 logger = logging.getLogger('django')
 
 
-class TimezoneMiddleware(object):
+class TimezoneMiddleware(MiddlewareMixin):
 
     def process_request(self, request):
         if request.user.is_authenticated():
@@ -36,7 +41,7 @@ class TimezoneMiddleware(object):
             timezone.deactivate()
 
 
-class LastIPMiddleware(object):
+class LastIPMiddleware(MiddlewareMixin):
 
     def process_request(self, request):
         if not request.user.is_authenticated():
@@ -52,7 +57,7 @@ class LastIPMiddleware(object):
             .update(last_ip=last_ip))
 
 
-class LastSeenMiddleware(object):
+class LastSeenMiddleware(MiddlewareMixin):
 
     def process_request(self, request):
         if not request.user.is_authenticated():
@@ -69,7 +74,7 @@ class LastSeenMiddleware(object):
             .update(last_seen=timezone.now()))
 
 
-class ActiveUserMiddleware(object):
+class ActiveUserMiddleware(MiddlewareMixin):
 
     def process_request(self, request):
         if not request.user.is_authenticated():

